### PR TITLE
Replace historyId with initial page tracking

### LINKED ISSUE #9378: After refresh, browser back button is not working.
# Bug report

## Describe the bug

After refresh, browser back button is not working.

## To Reproduce

It even reproduces on now.sh!

1. Go to https://zeit.co/home
2. Click on top left "pricing" navigation button
3. Browser refresh with `cmd` + `r`
4. Browser back button -> It just stays still.

## Expected behavior

It should go back to '/home'

## Screenshots

Nov-11-2019 17-10-38

## System information

- OS: macOS Catalina 10.15
- Browser : Chrome(Version 78.0.3904.97 (Official Build) (64-bit)) and safari(Version 13.0.2 (15608.2.30.1.1)) both reproduces this issue.
- Version of Next.js: I checked the latest version also reproduces this issue.

## Additional context

I can also reproduce with this example repo, which uses the latest next.js

```
npx create-next-app --example hello-world hello-world-app
```

### PR DESCRIPTION
**Pull request by ijjk**
This replaces the unique ID implementation with just tracking the extra popState for the initial page. I couldn't recall all of the finickiness of this browser bug on the historyId PR so going with this approach should be safer

x-ref: #9015
fixes: #9378

### CODE REVIEW

File: packages/next/next-server/lib/router/router.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ export default class Router implements BaseRouter {
   _bps: BeforePopStateCallback | undefined
   events: MittEmitter
   _wrapApp: (App: ComponentType) => any
-  historyId: number
+  isInitialPg: boolean
```
Comment (Timer) [inline packages/next/next-server/lib/router/router.ts]:
```suggestion
  isSsr: boolean
```

File: packages/next/next-server/lib/router/router.ts [hunk 2]
```diff
@@ -127,8 +127,9 @@ export default class Router implements BaseRouter {
     this.sub = subscription
     this.clc = null
     this._wrapApp = wrapApp
-    // we use a historyId to enable ignoring invalid popstates
-    this.historyId = Math.random()
+    // make sure to ignore extra popState in safari on navigating
+    // back from external site
+    this.isInitialPg = true
```
Comment (Timer) [inline packages/next/next-server/lib/router/router.ts]:
```suggestion
    this.isSsr = true
```

File: packages/next/next-server/lib/router/router.ts [hunk 3]
```diff
@@ -170,7 +171,12 @@ export default class Router implements BaseRouter {
 
     // Make sure we don't re-render on initial load,
     // can be caused by navigating back from an external site
-    if (e.state.options && e.state.options.historyId !== this.historyId) {
+    if (
+      e.state &&
+      this.isInitialPg &&
```
Comment (Timer) [inline packages/next/next-server/lib/router/router.ts]:
```suggestion
      this.isSsr &&
```

File: packages/next/next-server/lib/router/router.ts [hunk 4]
```diff
@@ -245,6 +251,7 @@ export default class Router implements BaseRouter {
 
   change(method: string, _url: Url, _as: Url, options: any): Promise<boolean> {
     return new Promise((resolve, reject) => {
+      this.isInitialPg = false
```
Comment (Timer) [inline packages/next/next-server/lib/router/router.ts]:
```suggestion
      this.isSsr = false
```

**Review by Timer — CHANGES_REQUESTED** [review]

File: packages/next/next-server/lib/router/router.ts [hunk 5]
```diff
@@ -245,6 +251,7 @@ export default class Router implements BaseRouter {
 
   change(method: string, _url: Url, _as: Url, options: any): Promise<boolean> {
     return new Promise((resolve, reject) => {
+      this.isSsr = false
```
Comment (Timer) [inline packages/next/next-server/lib/router/router.ts]:
```suggestion
      if (!options._h) {
        this.isSsr = false
      }
```

**Review by Timer — CHANGES_REQUESTED** [review]
We should not flip this flag until a real navigation occurs (instead of hydration reconciliation).

File: test/integration/client-navigation/test/index.test.js [hunk 6]
```diff
@@ -74,6 +74,18 @@ describe('Client Navigation', () => {
       await browser.close()
     })
 
+    it('should navigate back after reload', async () => {
+      const browser = await webdriver(context.appPort, '/nav')
+      await browser.elementByCss('#about-link').click()
+      await browser.waitForElementByCss('.nav-about')
+      await browser.eval(`window.location.href = window.location.href`)
+      await waitFor(500)
```
Comment (Timer) [inline test/integration/client-navigation/test/index.test.js]:
```suggestion
      await waitFor(5000)
```

File: test/integration/client-navigation/test/index.test.js [hunk 7]
```diff
@@ -74,6 +74,18 @@ describe('Client Navigation', () => {
...
+      await browser.eval(`window.history.back()`)
+      await waitFor(200)
```
Comment (Timer) [inline test/integration/client-navigation/test/index.test.js]:
```suggestion
      await waitFor(3000)
```

**Review by Timer — CHANGES_REQUESTED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| buildDuration | 13.2s | 13.5s | ⚠️ +247ms |
| nodeModulesSize | 48.8 MB | 48.8 MB | ⚠️ +164 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| main-HASH.js | 14.9 kB | 14.9 kB | ✓ |
| main-HASH.js gzip | 5.12 kB | 5.12 kB | ✓ |
| webpack-HASH.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..b3f403e8a.js | 19.8 kB | N/A | N/A |
| 16b1a7225520..3e8a.js gzip | 7.28 kB | N/A | N/A |
| 314e4c089430..de6eb480f.js | 7.9 kB | 7.9 kB | ✓ |
| 314e4c089430..480f.js gzip | 3.31 kB | 3.31 kB | ✓ |
| commons.HASH.js | 10.4 kB | 10.4 kB | ✓ |
| commons.HASH.js gzip | 3.91 kB | 3.91 kB | ✓ |
| de003c3a9d30..39988e4af.js | 33.1 kB | 33.1 kB | ✓ |
| de003c3a9d30..e4af.js gzip | 12.3 kB | 12.3 kB | ✓ |
| framework.HASH.js | 125 kB | 125 kB | ✓ |
| framework.HASH.js gzip | 39.4 kB | 39.4 kB | ✓ |
| 16b1a7225520..14b571e4d.js | N/A | 19.8 kB | N/A |
| 16b1a7225520..1e4d.js gzip | N/A | 7.28 kB | N/A |
| Overall change | 213 kB | 213 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| main-HASH.module.js | 14.1 kB | 14.1 kB | ✓ |
| main-HASH.module.js gzip | 5.31 kB | 5.31 kB | ✓ |
| webpack-HASH.module.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..d8.module.js | 16 kB | N/A | N/A |
| 16b1a7225520..dule.js gzip | 6.17 kB | N/A | N/A |
| 4952ddcd88e7..cc.module.js | 11.9 kB | 11.9 kB | ✓ |
| 4952ddcd88e7..dule.js gzip | 4.43 kB | 4.43 kB | ✓ |
| de003c3a9d30..36.module.js | 22.2 kB | 22.2 kB | ✓ |
| de003c3a9d30..dule.js gzip | 8.71 kB | 8.71 kB | ✓ |
| framework.HASH.module.js | 125 kB | 125 kB | ✓ |
| framework.HA..dule.js gzip | 39.4 kB | 39.4 kB | ✓ |
| 16b1a7225520..15.module.js | N/A | 16 kB | N/A |
| 16b1a7225520..dule.js gzip | N/A | 6.17 kB | N/A |
| Overall change | 191 kB | 191 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| polyfills-HASH.js | 9.11 kB | 9.11 kB | ✓ |
| polyfills-HASH.js gzip | 3.11 kB | 3.11 kB | ✓ |
| Overall change | 9.11 kB | 9.11 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| _app.js | 7.71 kB | 7.71 kB | ✓ |
| _app.js gzip | 3.22 kB | 3.22 kB | ✓ |
| _error.js | 14.9 kB | 14.9 kB | ✓ |
| _error.js gzip | 5.57 kB | 5.57 kB | ✓ |
| hooks.js | 1.92 kB | 1.92 kB | ✓ |
| hooks.js gzip | 941 B | 941 B | ✓ |
| index.js | 318 B | 318 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js | 4.05 kB | 4.05 kB | ✓ |
| link.js gzip | 1.81 kB | 1.81 kB | ✓ |
| routerDirect.js | 411 B | 411 B | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js | 421 B | 421 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 29.7 kB | 29.7 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| _app.module.js | 1.54 kB | 1.54 kB | ✓ |
| _app.module.js gzip | 759 B | 759 B | ✓ |
| _error.module.js | 10.5 kB | 10.5 kB | ✓ |
| _error.module.js gzip | 4.26 kB | 4.26 kB | ✓ |
| hooks.module.js | 677 B | 677 B | ✓ |
| hooks.module.js gzip | 384 B | 384 B | ✓ |
| index.module.js | 292 B | 292 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js | 5.49 kB | 5.49 kB | ✓ |
| link.module.js gzip | 2.47 kB | 2.47 kB | ✓ |
| routerDirect.module.js | 399 B | 399 B | ✓ |
| routerDirect..dule.js gzip | 285 B | 285 B | ✓ |
| withRouter.module.js | 409 B | 409 B | ✓ |
| withRouter.m..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 19.3 kB | 19.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| _buildManifest.js | 353 B | 353 B | ✓ |
| _buildManifest.js gzip | 259 B | 260 B | ⚠️ +1 B |
| _buildManifest.module.js | 251 B | 251 B | ✓ |
| _buildManife..dule.js gzip | 209 B | 210 B | ⚠️ +1 B |
| Overall change | 604 B | 604 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| index.html | 3.9 kB | 3.9 kB | ✓ |
| index.html gzip | 1.01 kB | 1.01 kB | ✓ |
| link.html | 4.58 kB | 4.58 kB | ✓ |
| link.html gzip | 1.14 kB | 1.14 kB | -1 B |
| withRouter.html | 4.43 kB | 4.43 kB | ✓ |
| withRouter.html gzip | 1.09 kB | 1.08 kB | -1 B |
| Overall change | 12.9 kB | 12.9 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>_buildManifest.js</strong></summary>

```diff
@@ -1 +1 @@
-self.__BUILD_MANIFEST = (function(a,b){return {"/_error":[b],"/link":[a,b],"/routerDirect":[a],"/withRouter":[a]}}("static\u002Fchunks\u002F16b1a72255206b7853bf6603dc58ec83de39b142.fa47d4aaa295dcae8d1b.js","static\u002Fchunks\u002F314e4c089430eb8cdc865a8a097be397fffdf2f3.dabd6bf1b89781ecd750.js"));self.__BUILD_MANIFEST_CB && self.__BUILD_MANIFEST_CB()
\ No newline at end of file
+self.__BUILD_MANIFEST = (function(a,b){return {"/_error":[b],"/link":[a,b],"/routerDirect":[a],"/withRouter":[a]}}("static\u002Fchunks\u002F16b1a72255206b7853bf6603dc58ec83de39b142.b3fcba13e968d4cbea5c.js","static\u002Fchunks\u002F314e4c089430eb8cdc865a8a097be397fffdf2f3.dabd6bf1b89781ecd750.js"));self.__BUILD_MANIFEST_CB && self.__BUILD_MANIFEST_CB()
\ No newline at end of file
```
</details>
<details>
<summary>Diff for <strong>_buildManifest.module.js</strong></summary>

```diff
@@ -1 +1 @@
-self.__BUILD_MANIFEST = (function(a){return {"/link":[a],"/routerDirect":[a],"/withRouter":[a]}}("static\u002Fchunks\u002F16b1a72255206b7853bf6603dc58ec83de39b142.59a07c7ab59f0e899509.module.js"));self.__BUILD_MANIFEST_CB && self.__BUILD_MANIFEST_CB()
\ No newline at end of file
+self.__BUILD_MANIFEST = (function(a){return {"/link":[a],"/routerDirect":[a],"/withRouter":[a]}}("static\u002Fchunks\u002F16b1a72255206b7853bf6603dc58ec83de39b142.684a61d8405dcae9572f.module.js"));self.__BUILD_MANIFEST_CB && self.__BUILD_MANIFEST_CB()
\ No newline at end of file
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -16,7 +16,7 @@
         as="script" crossorigin="anonymous" />
         <link rel="preload" href="/_next/static/chunks/de003c3a9d308750aa009870a5926f9b18ab31f4.dda3a797bb804537ed85.module.js"
         as="script" crossorigin="anonymous" />
-        <link rel="preload" href="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.59a07c7ab59f0e899509.module.js"
+        <link rel="preload" href="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.684a61d8405dcae9572f.module.js"
         as="script" crossorigin="anonymous" />
         <link rel="preload" href="/_next/static/chunks/4952ddcd88e7185e66c9cf40e2d848b7e27f1574.6ad24cd530d796126b06.module.js"
         as="script" crossorigin="anonymous" />
@@ -59,9 +59,9 @@
         async="" crossorigin="anonymous" nomodule=""></script>
         <script src="/_next/static/chunks/de003c3a9d308750aa009870a5926f9b18ab31f4.dda3a797bb804537ed85.module.js"
         async="" crossorigin="anonymous" type="module"></script>
-        <script src="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.fa47d4aaa295dcae8d1b.js"
+        <script src="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.b3fcba13e968d4cbea5c.js"
         async="" crossorigin="anonymous" nomodule=""></script>
-        <script src="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.59a07c7ab59f0e899509.module.js"
+        <script src="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.684a61d8405dcae9572f.module.js"
         async="" crossorigin="anonymous" type="module"></script>
         <script src="/_next/static/chunks/314e4c089430eb8cdc865a8a097be397fffdf2f3.dabd6bf1b89781ecd750.js"
         async="" crossorigin="anonymous" nomodule=""></script>
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -16,7 +16,7 @@
         as="script" crossorigin="anonymous" />
         <link rel="preload" href="/_next/static/chunks/de003c3a9d308750aa009870a5926f9b18ab31f4.dda3a797bb804537ed85.module.js"
         as="script" crossorigin="anonymous" />
-        <link rel="preload" href="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.59a07c7ab59f0e899509.module.js"
+        <link rel="preload" href="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.684a61d8405dcae9572f.module.js"
         as="script" crossorigin="anonymous" />
         <link rel="preload" href="/_next/static/chunks/4952ddcd88e7185e66c9cf40e2d848b7e27f1574.6ad24cd530d796126b06.module.js"
         as="script" crossorigin="anonymous" />
@@ -57,9 +57,9 @@
         async="" crossorigin="anonymous" nomodule=""></script>
         <script src="/_next/static/chunks/de003c3a9d308750aa009870a5926f9b18ab31f4.dda3a797bb804537ed85.module.js"
         async="" crossorigin="anonymous" type="module"></script>
-        <script src="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.fa47d4aaa295dcae8d1b.js"
+        <script src="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.b3fcba13e968d4cbea5c.js"
         async="" crossorigin="anonymous" nomodule=""></script>
-        <script src="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.59a07c7ab59f0e899509.module.js"
+        <script src="/_next/static/chunks/16b1a72255206b7853bf6603dc58ec83de39b142.684a61d8405dcae9572f.module.js"
         async="" crossorigin="anonymous" type="module"></script>
         <script src="/_next/static/chunks/4952ddcd88e7185e66c9cf40e2d848b7e27f1574.6ad24cd530d796126b06.module.js"
         async="" crossorigin="anonymous" type="module"></script>
```
</details>
</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| buildDuration | 13.6s | 13.7s | ⚠️ +96ms |
| nodeModulesSize | 48.8 MB | 48.8 MB | ⚠️ +164 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| main-HASH.js | 14.9 kB | 14.9 kB | ✓ |
| main-HASH.js gzip | 5.12 kB | 5.12 kB | ✓ |
| webpack-HASH.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..b3f403e8a.js | 19.8 kB | N/A | N/A |
| 16b1a7225520..3e8a.js gzip | 7.28 kB | N/A | N/A |
| 314e4c089430..de6eb480f.js | 7.9 kB | 7.9 kB | ✓ |
| 314e4c089430..480f.js gzip | 3.31 kB | 3.31 kB | ✓ |
| commons.HASH.js | 10.4 kB | 10.4 kB | ✓ |
| commons.HASH.js gzip | 3.91 kB | 3.91 kB | ✓ |
| de003c3a9d30..39988e4af.js | 33.1 kB | 33.1 kB | ✓ |
| de003c3a9d30..e4af.js gzip | 12.3 kB | 12.3 kB | ✓ |
| framework.HASH.js | 125 kB | 125 kB | ✓ |
| framework.HASH.js gzip | 39.4 kB | 39.4 kB | ✓ |
| 16b1a7225520..14b571e4d.js | N/A | 19.8 kB | N/A |
| 16b1a7225520..1e4d.js gzip | N/A | 7.28 kB | N/A |
| Overall change | 213 kB | 213 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| main-HASH.module.js | 14.1 kB | 14.1 kB | ✓ |
| main-HASH.module.js gzip | 5.31 kB | 5.31 kB | ✓ |
| webpack-HASH.module.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..d8.module.js | 16 kB | N/A | N/A |
| 16b1a7225520..dule.js gzip | 6.17 kB | N/A | N/A |
| 4952ddcd88e7..cc.module.js | 11.9 kB | 11.9 kB | ✓ |
| 4952ddcd88e7..dule.js gzip | 4.43 kB | 4.43 kB | ✓ |
| de003c3a9d30..36.module.js | 22.2 kB | 22.2 kB | ✓ |
| de003c3a9d30..dule.js gzip | 8.71 kB | 8.71 kB | ✓ |
| framework.HASH.module.js | 125 kB | 125 kB | ✓ |
| framework.HA..dule.js gzip | 39.4 kB | 39.4 kB | ✓ |
| 16b1a7225520..15.module.js | N/A | 16 kB | N/A |
| 16b1a7225520..dule.js gzip | N/A | 6.17 kB | N/A |
| Overall change | 191 kB | 191 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| polyfills-HASH.js | 9.11 kB | 9.11 kB | ✓ |
| polyfills-HASH.js gzip | 3.11 kB | 3.11 kB | ✓ |
| Overall change | 9.11 kB | 9.11 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| _app.js | 7.71 kB | 7.71 kB | ✓ |
| _app.js gzip | 3.22 kB | 3.22 kB | ✓ |
| _error.js | 14.9 kB | 14.9 kB | ✓ |
| _error.js gzip | 5.57 kB | 5.57 kB | ✓ |
| hooks.js | 1.92 kB | 1.92 kB | ✓ |
| hooks.js gzip | 941 B | 941 B | ✓ |
| index.js | 318 B | 318 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js | 4.05 kB | 4.05 kB | ✓ |
| link.js gzip | 1.81 kB | 1.81 kB | ✓ |
| routerDirect.js | 411 B | 411 B | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js | 421 B | 421 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 29.7 kB | 29.7 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| _app.module.js | 1.54 kB | 1.54 kB | ✓ |
| _app.module.js gzip | 759 B | 759 B | ✓ |
| _error.module.js | 10.5 kB | 10.5 kB | ✓ |
| _error.module.js gzip | 4.26 kB | 4.26 kB | ✓ |
| hooks.module.js | 677 B | 677 B | ✓ |
| hooks.module.js gzip | 384 B | 384 B | ✓ |
| index.module.js | 292 B | 292 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js | 5.49 kB | 5.49 kB | ✓ |
| link.module.js gzip | 2.47 kB | 2.47 kB | ✓ |
| routerDirect.module.js | 399 B | 399 B | ✓ |
| routerDirect..dule.js gzip | 285 B | 285 B | ✓ |
| withRouter.module.js | 409 B | 409 B | ✓ |
| withRouter.m..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 19.3 kB | 19.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| _buildManifest.js | 353 B | 353 B | ✓ |
| _buildManifest.js gzip | 259 B | 260 B | ⚠️ +1 B |
| _buildManifest.module.js | 251 B | 251 B | ✓ |
| _buildManife..dule.js gzip | 209 B | 210 B | ⚠️ +1 B |
| Overall change | 604 B | 604 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary | ijjk/next.js fix/rm-historyId | Change |
| - | - | - | - |
| _error.js | 244 kB | 244 kB | ✓ |
| _error.js gzip | 64.9 kB | 64.9 kB | -2 B |
| hooks.html | 4.04 kB | 4.04 kB | ✓ |
| hooks.html gzip | 1.05 kB | 1.05 kB | ✓ |
| index.js | 244 kB | 244 kB | ✓ |
| index.js gzip | 65 kB | 65 kB | -2 B |
| link.js | 276 kB | 276 kB | ⚠️ +21 B |
| link.js gzip | 73.9 kB | 73.9 kB | ⚠️ +4 B |
| routerDirect.js | 269 kB | 269 kB | ⚠️ +21 B |
| routerDirect.js gzip | 72 kB | 72 kB | ⚠️ +7 B |
| withRouter.js | 269 kB | 269 kB | ⚠️ +21 B |
| withRouter.js gzip | 72.1 kB | 72.1 kB | ⚠️ +8 B |
| Overall change | 1.31 MB | 1.31 MB | ⚠️ +63 B |

</details>

</details>


Commit: 693d437a2c32ce1d4bdc4871cf789b811454d100

**Review by Timer — APPROVED** [review]

**Comment by Insteadof** [comment]
Thanks! I discovered a bug in the routing when refreshing my page and navigating back. I debugged it all the way to the line that went wrong and thankfully noticed a very recent commit on this line of code. So I installed the very latest version and my issue appears to be fixed! Thanks a lot.

**Pull request merged**